### PR TITLE
fix: silence TypeError during tear down stage

### PR DESCRIPTION
**Pull request by lidizheng**
Related https://github.com/googleapis/google-auth-library-python/issues/1016

This PR adds a workaround to silence the TypeError. The root cause of the issue sits in CPython, it's unlikely for us to remove this workaround any time soon.

### CODE REVIEW

**Comment by lidizheng** [comment]
@arithmetic1728 PTAL. The failed test cases don't look related to this PR.

**Comment by arithmetic1728** [comment]
@lidizheng Thank you! Yes, the credential for system test is expired, I will fix it.

**Review by arithmetic1728 — APPROVED** [review]

**Comment by arithmetic1728** [comment]
Will merge this PR after #1028 is merged.

**Pull request merged**
